clang/lib/Tooling/DependencyScanning/ModuleDepCollector.cpp

```diff
@@ -532,7 +539,8 @@ ModuleDepCollectorPP::handleTopLevelModule(const Module *M) {
   // Finish the compiler invocation. Requires dependencies and the context hash.
   MDC.addOutputPaths(CI, MD);
 
-  MD.BuildArguments = CI.getCC1CommandLine();
+  MD.BuildInvocationOrArguments =
+      std::make_unique<CowCompilerInvocation>(std::move(CI));
```

Comment:
You have shared_ptr but call make_unique; it works but seems unexpected

Reply:
Good catch, I'll align it with whatever pointer/value type we decide to go with.